seqr/views/utils/dataset_utils.py

```diff
@@ -264,29 +204,38 @@ def update_variant_samples(samples, user, elasticsearch_index, data_source=None,
     return activated_sample_guids, inactivate_sample_guids
 
 
-def match_and_update_samples(
-        projects, user, sample_ids, sample_type, elasticsearch_index=None, data_source=None, dataset_type=Sample.DATASET_TYPE_VARIANT_CALLS,
-        sample_id_to_individual_id_mapping=None, raise_no_match_error=False,
-        raise_unmatched_error_template=None, allow_partial_families=False,
+def match_and_update_search_samples(
+        project, user, sample_ids, elasticsearch_index, sample_type, dataset_type,
+        sample_id_to_individual_id_mapping, raise_unmatched_error_template, expected_families=None,
 ):
+    samples = Sample.objects.select_related('individual').filter(
+        individual__family__project=project,
+        sample_type=sample_type,
+        dataset_type=dataset_type,
+        sample_id__in=sample_ids,
+        elasticsearch_index=elasticsearch_index,
+    )
     loaded_date = timezone.now()
-    samples, included_families, matched_individual_ids, remaining_sample_ids = match_sample_ids_to_sample_records(
-        projects=projects,
+    samples, matched_individual_ids, remaining_sample_ids = _find_or_create_missing_sample_records(
+        samples=samples,
+        projects=[project],
         user=user,
         sample_ids=sample_ids,
         elasticsearch_index=elasticsearch_index,
-        data_source=data_source,
         sample_type=sample_type,
         dataset_type=dataset_type,
         sample_id_to_individual_id_mapping=sample_id_to_individual_id_mapping,
         loaded_date=loaded_date,
-        raise_no_match_error=raise_no_match_error,
+        raise_no_match_error=not raise_unmatched_error_template,
         raise_unmatched_error_template=raise_unmatched_error_template,
-        allow_partial_families=allow_partial_families,
     )
 
-    activated_sample_guids, inactivated_sample_guids = update_variant_samples(
-        samples, user, elasticsearch_index, data_source, loaded_date, dataset_type, sample_type)
+    prefetch_related_objects(samples, 'individual__family')
+    included_families = {sample.individual.family for sample in samples}
+    _validate_samples_families(samples, included_families, sample_type, dataset_type, expected_families=expected_families)
+
+    activated_sample_guids, inactivated_sample_guids = _update_variant_samples(
+        samples, user, elasticsearch_index, loaded_date=loaded_date, dataset_type=dataset_type, sample_type=sample_type)
```

Comment:
It is okay to call with positional arguments rather than keyword arguments.

Reply:
yes